### PR TITLE
`Element.ShadowRoot` returns error when `ShadowRoots` is empty

### PR DESCRIPTION
**Pull request by pcen**
Addresses #787 

# Development guide

[Link](https://github.com/go-rod/rod/blob/master/.github/CONTRIBUTING.md)

## Test on local before making the PR

```bash
go run ./lib/utils/simple-check
```

### CODE REVIEW

File: element.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ import (
 	"github.com/go-rod/rod/lib/utils"
 )
 
+var (
+	// ErrNoShadowRoot is calling ShadowRoot on an Element with no shadow root
+	ErrNoShadowRoot = errors.New("element has no shadow root")
```
Comment (ysmood) [inline element.go:22]: Please move it to https://github.com/go-rod/rod/blob/master/error.go

**Review by ysmood — CHANGES_REQUESTED** [review]
Please add a unit test for it. Check the development guide for how to test the project.

**Review by ysmood — APPROVED** [review]
God job!

**Review by ysmood — CHANGES_REQUESTED** [review]
Please fix the ci coverage error. 

If you don't how please read: https://github.com/go-rod/rod/blob/master/.github/CONTRIBUTING.md

**Comment by pcen** [comment]
merging should close https://github.com/go-rod/rod/issues/787

**Review by ysmood — APPROVED** [review]

**Pull request merged**